lib/ocpp/v201/monitoring_updater.cpp

```diff
@@ -0,0 +1,532 @@
+                PeriodicMonitorData periodic_monitor_data;
+
+                periodic_monitor_data.component = component_variable_monitors.component;
+                periodic_monitor_data.variable = component_variable_monitors.variable;
+                periodic_monitor_data.monitor_meta = periodic_monitor_meta;
+
+                if (periodic_monitor_meta.monitor.type == MonitorEnum::Periodic) {
+                    // Set the trigger to the current time
+                    periodic_monitor_data.last_trigger_steady = std::chrono::steady_clock::now();
+                } else if (periodic_monitor_meta.monitor.type == MonitorEnum::PeriodicClockAligned) {
+                    // Snap to the closest monitor multiple
+                    periodic_monitor_data.next_trigger_clock_aligned =
+                        get_next_clock_aligned_point(periodic_monitor_data.monitor_meta.monitor.value);
+                    EVLOG_debug << "First aligned timepoint for monitor ID: " << periodic_monitor_meta.monitor.id;
+                }
+
+                auto res = this->periodic_monitors.insert(
+                    std::pair{periodic_monitor_meta.monitor.id, std::move(periodic_monitor_data)});
+
+                if (!res.second) {
+                    EVLOG_warning << "Could not insert monitor to periodic monitor map!";
+                    continue;
+                }
+
+                it = res.first;
+            }
+
+            PeriodicMonitorData& periodic_monitor = it->second;
+            bool matches_time = false;
+
+            if (periodic_monitor.monitor_meta.monitor.type == MonitorEnum::Periodic) {
+                auto current_time = std::chrono::steady_clock::now();
+                auto delta = current_time - periodic_monitor.last_trigger_steady;
+
+                if (delta > monitor_seconds) {
+                    // Update last time
+                    periodic_monitor.last_trigger_steady = current_time;
+                    matches_time = true;
+                }
+            } else if (periodic_monitor.monitor_meta.monitor.type == MonitorEnum::PeriodicClockAligned) {
+                // 3.55
+                // PeriodicClockAligned Triggers an event notice every monitorValue
+                // seconds interval, starting from the nearest clock-aligned interval
+                // after this monitor was set. For example, a monitorValue of 900 will
+                // trigger event notices at 0, 15, 30 and 45 minutes after the hour, every hour.
+                auto current_time = std::chrono::system_clock::now();
+
+                if (current_time > periodic_monitor.next_trigger_clock_aligned) {
+                    auto distance = std::chrono::duration_cast<std::chrono::seconds>(
+                                        current_time - periodic_monitor.next_trigger_clock_aligned)
+                                        .count();
+
+                    // TODO (ioan): if we missed an interval, for example 15 minute mark see how large
```

Comment:
is that still a TODO?
How can we miss `distance` become greater than 60 when we run `process_periodic_monitors` every second?

Reply:
Will be handled in a future PR with: N08.FR.03